rudder-lang/src/parser/error.rs

```diff
@@ -123,27 +124,44 @@ impl<I: Clone> ParseError<I> for PError<I> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let message = match &self.kind {
-            PErrorKind::Nom(e) => format!("Unprocessed parsing error: {:?}.\nPlease fill a BUG with context on when this happened!", e),
+            PErrorKind::Nom(e) => format!("Unprocessed parsing error: '{:#?}'.\nPlease fill a BUG with context on when this happened!", e),
             #[cfg(test)]
             PErrorKind::NomTest(msg) => format!("Testing only error message, this should never happen {}.\nPlease fill a BUG with context on when this happened!", msg),
-            PErrorKind::InvalidFormat => "Invalid header format, it must contain a single line '@format=x' where x is an integer. Shebang accepted.".to_string(),
-            PErrorKind::InvalidName(i) => format!("The identifier is invalid in a {}.", i.fragment.bright_magenta()),
-            PErrorKind::UnexpectedToken(s) => format!("Unexpected token, expecting '{}', found", s.bright_magenta()),
-            PErrorKind::UnterminatedDelimiter(i) => format!("Missing closing delimiter for '{}'", i.fragment.bright_magenta()),
+            PErrorKind::ExpectedKeyword(s) => format!("The following value kind was expected: '{}'.", s.bright_magenta()),
+            PErrorKind::ExpectedReservedWord(s) => format!("The following reserved keyword was expected: '{}'.", s.bright_magenta()),
+            PErrorKind::ExpectedToken(s) => format!("The following token was expected '{}'.", s.bright_magenta()),
             PErrorKind::InvalidEnumExpression => "This enum expression is invalid".to_string(),
             PErrorKind::InvalidEscapeSequence => "This escape sequence cannot be used in a string".to_string(),
+            PErrorKind::InvalidFormat => "Invalid header format, it must contain a single line '@format=x' where x is an integer. Shebang accepted.".to_string(),
+            PErrorKind::InvalidName(i) => format!("The identifier is invalid in a {}.", i.fragment.bright_magenta()),
             PErrorKind::InvalidVariableReference => "This variable reference is invalid".to_string(),
-            PErrorKind::ExpectedKeyword(s) => format!("Token not found, expected '{}'", s.bright_magenta()),
             PErrorKind::UnsupportedMetadata(i) => format!("Parsed comment or metadata not supported at this place: '{}' fount at {}", i.fragment.bright_magenta(), Token::from(*i).position_str().bright_yellow()),
+            PErrorKind::UnterminatedDelimiter(i) => format!("Missing closing delimiter for '{}'", i.fragment.bright_magenta()),
         };
-        f.write_str(&format!(
-            "{}, near\n{}{} {}",
-            Token::from(self.context).position_str().bright_yellow(),
-            self.context.fragment,
-            "-->".bright_blue(),
-            message.bold(),
-            
-        ))
+
+        // simply removes superfluous line return (prettyfication)
+        match self.context {
+            Some(ctx) => {
+                let mut context = ctx.fragment.to_owned();
```

Comment:
Using https://doc.rust-lang.org/std/primitive.str.html#method.trim_end_matches would avoid creating a new mutable string, but it's not mandatory and current solution is acceptable.

Reply:
nice tip, thanks, updated